app/actions/websocket/channel.ts

```diff
@@ -18,42 +27,201 @@ import {getCurrentUser, getUserById} from '@queries/servers/user';
+// Received when current user created a channel in a different client
+export async function handleChannelCreatedEvent(serverUrl: string, msg: any) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
         return;
     }
-    const currentUser = await getCurrentUser(database.database);
-    const {team_id: teamId, user_id: userId} = msg.data;
-    const {channel_id: channelId} = msg.broadcast;
-    const models: Model[] = [];
+
+    const {team_id: teamId, channel_id: channelId} = msg.data;
 
     try {
-        const addedUser = getUserById(database.database, userId);
-        if (!addedUser) {
-            // TODO Potential improvement https://mattermost.atlassian.net/browse/MM-40581
-            const {users} = await fetchUsersByIds(serverUrl, [userId], true);
-            if (users) {
-                models.push(...await database.operator.handleUsers({users, prepareRecordsOnly: true}));
+        const models: Model[] = [];
+        const {channels, memberships} = await fetchMyChannel(serverUrl, teamId, channelId, true);
+        if (channels && memberships) {
+            const prepare = await prepareMyChannelsForTeam(operator, teamId, channels, memberships);
+            if (prepare) {
+                const prepareModels = await Promise.all(prepare);
+                const flattenedModels = prepareModels.flat();
+                if (flattenedModels?.length > 0) {
+                    models.push(...flattenedModels);
+                }
+                const categoryModels = await addChannelToDefaultCategory(serverUrl, channels[0], true);
+                if (categoryModels.models?.length) {
+                    models.push(...categoryModels.models);
+                }
             }
         }
+        operator.batchRecords(models);
+    } catch {
+        // do nothing
+    }
+}
+
+export async function handleChannelUnarchiveEvent(serverUrl: string, msg: any) {
+    try {
+        await setChannelDeleteAt(serverUrl, msg.data.channel_id, 0);
+    } catch {
+        // do nothing
+    }
+}
+
+export async function handleChannelConvertedEvent(serverUrl: string, msg: any) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return;
+    }
+
+    try {
+        const channelId = msg.data.channel_id;
+        const {channel} = await fetchChannelById(serverUrl, channelId);
+        if (channel) {
+            operator.handleChannel({channels: [channel], prepareRecordsOnly: false});
+        }
+    } catch {
+        // do nothing
+    }
+}
+
+export async function handleChannelUpdatedEvent(serverUrl: string, msg: any) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return;
+    }
+
+    const updatedChannel = JSON.parse(msg.data.channel);
+    try {
+        const models: Model[] = await operator.handleChannel({channels: [updatedChannel], prepareRecordsOnly: true});
```

Comment:
should we check that we have the channel in the database to avoid it from being created?

Reply:
We could, but I wonder what scenario would that be. In theory, if we are receiving this websocket event is because we are members of that channel. Therefore, we should be updating it, and if it is not in the database, adding it, right?